spec/index.bs

```diff
@@ -8025,10 +8103,13 @@ GPUSwapChain includes GPUObjectBase;
     To <dfn abstract-op lt='allocating a new swap chain texture'>Allocate a new swap chain texture</dfn>
     for {{GPUSwapChain}} |swapChain| run the following steps:
 
-        1. Let |canvas| be |swapChain|.{{GPUSwapChain/[[context]]}}.{{GPUCanvasContext/[[canvas]]}}.
+        1. If |swapChain| is [=invalid=]:
+            1. Generate a {{GPUValidationError}} in the current scope with an appropriate error message.
+            1. Return a new [=invalid=] {{GPUTexture}}.
         1. Let |device| be |swapChain|.{{GPUSwapChain/[[descriptor]]}}.{{GPUSwapChainDescriptor/device}}.
         1. Let |descriptor| be a new {{GPUTextureDescriptor}}.
-        1. Set |descriptor|.{{GPUTextureDescriptor/size}} to [|canvas|.width, |canvas|.height, 1].
+        1. Set |descriptor|.{{GPUTextureDescriptor/size}} to
+            |swapChain|.{{GPUSwapChain/[[descriptor]]}}.{{GPUSwapChainDescriptor/size}}.
```

Comment:
similarly here, is it valid to set this, given that the right hand side is nullable?